### PR TITLE
Restore email registration login functionality

### PR DESCRIPTION
**Pull request by jsavell**
AppLoginController.js needs to extend  weaver LoginController, not AuthenticationController or the email login functionality is broken due to the different login method implementations.

https://github.com/TAMULib/Weaver-UI-Core/blob/2.x/app/controllers/authenticationController.js#L54

https://github.com/TAMULib/Weaver-UI-Core/blob/2.x/app/controllers/loginController.js#L7

### CODE REVIEW

**Comment by wwelling** [comment]
The exact opposite was a PR during CAP maintenance. https://github.com/TAMULib/CAP/commit/c196258f988f3a2fb9748d20a8cc8125a16f22ab#diff-6b728c506f53d8487647a783983a5889

**Comment by wwelling** [comment]
This PR on SAGE followed the CAP PR. 

https://github.com/TAMULib/SAGE/commit/4c421196819c7f6bd44f8d7c211e6952ea8800e1#diff-6b728c506f53d8487647a783983a5889

It appears we will have to remedy this as well. I think it may be ideal to distinguish the scope function for each type of login.

**Comment by wwelling** [comment]
Is it possible, that this change back to the LoginController breaks CAS login when allowing anonymous?

**Comment by jsavell** [comment]
> 
> 
> The exact opposite was a PR during CAP maintenance. [TAMULib/CAP@c196258#diff-6b728c506f53d8487647a783983a5889](https://github.com/TAMULib/CAP/commit/c196258f988f3a2fb9748d20a8cc8125a16f22ab#diff-6b728c506f53d8487647a783983a5889)
> 
> It appears one is for basic login and another is to invoke CAS login.

The same change is here in SAGE, too, which suggests CAP email registration is also broken right now.

https://github.com/TAMULib/SAGE/commit/4c421196819c7f6bd44f8d7c211e6952ea8800e1

**Comment by jsavell** [comment]
The appLoginController class was unchanged from initial implementation until the change to extend AuthenticationController, so if anonymous CAS login is somehow broken, it needs to be solved in a way that doesn't break email registration.

**Comment by wwelling** [comment]
Unless allow anonymous CAS login was implemented and working prior to addition of basic login. Then basic login needs to not break allow anonymous CAS login. Allow anonymous is configurable in appConfig.js. I just simply don't know what worked and when. Therefor, don't know what broke what.

File: src/main/webapp/app/controllers/appLoginController.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 sage.controller('AppLoginController', function ($controller, $scope) {
 
-  angular.extend(this, $controller('AuthenticationController', {
+  angular.extend(this, $controller('LoginController', {
```
Comment (wwelling) [inline src/main/webapp/app/controllers/appLoginController.js:3]: This is correct as AppLoginController is intended to extend weaver LoginController. However, this leaves external login broken. The following changes to the loginModal.html are needed:

remove ng-controller="AppLoginController" from the top level and add it to the form here:
https://github.com/TAMULib/SAGE/blob/master/src/main/webapp/app/views/modals/loginModal.html#L9

add ng-controller="AuthenticationController" here:
https://github.com/TAMULib/SAGE/blob/master/src/main/webapp/app/views/modals/loginModal.html#L37

**Pull request merged**